### DISCUSSION TITLE
Point domain name at home page of an app

### DISCUSSION THREAD

**Discussion by rpmcoding** [Budibase/budibase#7939] [Help] [answered]
Currently I have several apps at 

domain.com/app/app1
domain.com/app/app2
domain.com/app/app3

etc.

I would like to point 

newdomain1.com to what you currently see on domain.com/app/app1 home page
newdomain2.com to what you currently see on domain.com/app/app2 home page
newdomain3.com to what you currently see on domain.com/app/app3 home page

Does anyone know how this would be done?  Thanks in advance.

**Comment by melohagan** [comment, marked as answer]
Hey @rpmcoding 

There's some nginx docs here that may be useful: https://docs.budibase.com/docs/reverse-proxy

**Reply by rpmcoding** [reply]
@melohagan yes I did use those instructions to set up the domain.  My question however was more around how to point multiple different domains at different specific apps.

If I had say seven apps and seven different domain names for them.

`server {
    listen 80;
    server_name yourdomain.com;
    
    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $http_host;
        proxy_http_version 1.1;
        proxy_pass http://localhost:10000;
    }
    
    client_max_body_size 50m;
}`

I tried updating proxy_pass http://localhost:10000; to proxy_pass http://localhost:10000/apps/{appname}; but it did not work.  Any ideas?

**Reply by shogunpurple** [reply]
Hi @rpmcoding, this config looks ok, how are you running NGINX? Did you restart your NGINX after making changes?

**Reply by rpmcoding** [reply]
@shogunpurple yes that config is ok for 1 domain.  What I am trying to achieve though is multiple domains pointing at multiple app home pages.

So for example say I currently have an app homepage at https://domain.com/apps/apples and another app homepage at https://domain.com/apps/bananas.

But then I want to use 2 new domain names to display these apps public homepages (not the builder login).

e.g. I want https://apples.com to render https://domain.com/apps/apples and https://bananas.com to render https://domain.com/apps/bananas

etc etc.

I just can't see how to achieve that...?

**Reply by rpmcoding** [reply]
@shogunpurple any ideas on how this can be accomplished?

**Reply by shogunpurple** [reply]
you will need to replicate your `server` block to serve from the different domains:

```
  server {
    listen       10000 domain1.net;

...

  server {
    listen       10000 domain2.net;

```

This will show you what you need to do.
https://www.nginx.com/resources/wiki/start/topics/examples/server_blocks/

Not really budibase specific so marking this as answer.

**Reply by rpmcoding** [reply]
@shogunpurple thanks.  

I had previously set up the server blocks so that I can serve budibase from multiple domains and that worked just fine.  

The part I am struggling with is pointing a domain or subdomain in this case to the deployed app's home page rather than the budibase builder login screen.

When accessing ila.domain.com I get the error:- "There was an error loading your app
The Budibase client library could not be loaded. Try republishing your app."

Here is the server block code:-

```
server {
    server_name ila.domain.com;

    location / {
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header Host $http_host;
        proxy_http_version 1.1;
        proxy_pass http://localhost:10000/app/ila;
    }

    client_max_body_size 50m;

    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/ila.domain.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/ila.domain.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}

server {
    if ($host = ila.domain.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

    server_name ila.domain.com;
    listen 80;
    return 404; # managed by Certbot
}
```

**Reply by JayP718** [reply]
Hi did you ever get this working and if so how? @rpmcoding

**Reply by rpmcoding** [reply]
Unfortunately not...

**Reply by IchthysMaranatha** [reply]
Can be done this way, only showing key changes needed
	location / {

rewrite ^(/.+)$ $1 break;
 rewrite  ^(.*)$ /app/ila;
.....
.....
		proxy_pass http://localhost:10000;
	}
Now landing on the domain or subdomain which has been set up, it will directly go to the app
